k3k-kubelet/provider/provider.go

```diff
@@ -270,46 +446,44 @@ func (p *Provider) GetPodStatus(ctx context.Context, namespace, name string) (*c
 // concurrently outside of the calling goroutine. Therefore it is recommended
 // to return a version after DeepCopy.
 func (p *Provider) GetPods(ctx context.Context) ([]*corev1.Pod, error) {
-	labelSelector := metav1.AddLabelToSelector(&metav1.LabelSelector{}, clusterNameLabel, p.ClusterName)
-	pods, err := p.CoreClient.Pods(p.ClusterNamespace).List(ctx, metav1.ListOptions{LabelSelector: labelSelector.String()})
+	selector := labels.NewSelector()
+	requirement, err := labels.NewRequirement(clusterNameLabel, selection.Equals, []string{p.ClusterName})
 	if err != nil {
-		return nil, err
+		return nil, fmt.Errorf("unable to create label selector: %w", err)
+	}
+	selector = selector.Add(*requirement)
+	var podList corev1.PodList
+	err = p.HostClient.List(ctx, &podList, &client.ListOptions{LabelSelector: selector})
+	if err != nil {
+		return nil, fmt.Errorf("unable to list pods: %w", err)
 	}
 	retPods := []*corev1.Pod{}
-	for _, pod := range pods.DeepCopy().Items {
-		retPods = append(retPods, p.translateFrom(&pod))
+	for _, pod := range podList.DeepCopy().Items {
+		p.Translater.TranslateFrom(&pod)
+		retPods = append(retPods, &pod)
 	}
 	return retPods, nil
 }
 
-// translateTo translates a pod's definition from the virutal cluster to the definition on the host cluster
-func (p *Provider) translateTo(virtualPod *corev1.Pod) *corev1.Pod {
-	hostPod := virtualPod.DeepCopy()
-	hostPod.Name = p.hostName(virtualPod.Namespace, virtualPod.Name)
-	hostPod.Namespace = p.ClusterNamespace
-	if hostPod.Labels == nil {
-		hostPod.Labels = map[string]string{}
-	}
-	if hostPod.Annotations == nil {
-		hostPod.Annotations = map[string]string{}
-	}
-	hostPod.Labels[clusterNameLabel] = p.ClusterName
-	hostPod.Annotations[podNameAnnotation] = virtualPod.Name
-	hostPod.Annotations[podNamespaceAnnotation] = virtualPod.Namespace
-	return hostPod
-}
-
-// translateFrom translates a pod's definition from the host cluster to the definition on the host cluster
-func (p *Provider) translateFrom(hostPod *corev1.Pod) *corev1.Pod {
-	virtualPod := hostPod.DeepCopy()
-	delete(virtualPod.Labels, clusterNameLabel)
-	virtualPod.Name = hostPod.Annotations[podNameAnnotation]
-	virtualPod.Namespace = hostPod.Annotations[podNamespaceAnnotation]
-	delete(virtualPod.Annotations, podNameAnnotation)
-	delete(virtualPod.Annotations, podNamespaceAnnotation)
-	return virtualPod
-}
-
-func (p *Provider) hostName(virtualNamespace, virtualName string) string {
-	return controller.SafeConcatName(p.ClusterName, p.ClusterNamespace, virtualNamespace, virtualName)
+// getSecretsAndConfigmaps retrieves a list of all secrets/configmaps that are in use by a given pod. Useful
+// for removing/seeing which virtual cluster resources need to be in the host cluster.
+func getSecretsAndConfigmaps(pod *corev1.Pod) ([]string, []string) {
+	secrets := []string{}
```

Comment:
var secrets []string for here and below. This is the standard the repo is putting forth.